pkgs/os-specific/linux/lvm2/no-dynamic.patch

```diff
@@ -0,0 +1,267 @@
+--- a/configure
++++ b/configure
+@@ -659,6 +659,7 @@ STATIC_LINK
+ STATICDIR
+ SNAPSHOTS
+ SYSCONFDIR
++SHARED_LINK
+ SELINUX_PC
+ SELINUX_LIBS
+ SBINDIR
+@@ -866,6 +867,7 @@ infodir
+ docdir
+ oldincludedir
+ includedir
++runstatedir
+ localstatedir
+ sharedstatedir
+ sysconfdir
+@@ -891,6 +893,7 @@ enable_option_checking
+ enable_dependency_tracking
+ enable_silent_rules
+ enable_static_link
++enable_shared
+ with_user
+ with_group
+ with_device_uid
+@@ -1054,6 +1057,7 @@ datadir='${datarootdir}'
+ sysconfdir='${prefix}/etc'
+ sharedstatedir='${prefix}/com'
+ localstatedir='${prefix}/var'
++runstatedir='${localstatedir}/run'
+ includedir='${prefix}/include'
+ oldincludedir='/usr/include'
+ docdir='${datarootdir}/doc/${PACKAGE}'
+@@ -1306,6 +1310,15 @@ do
+   | -silent | --silent | --silen | --sile | --sil)
+     silent=yes ;;
+ 
++  -runstatedir | --runstatedir | --runstatedi | --runstated \
++  | --runstate | --runstat | --runsta | --runst | --runs \
++  | --run | --ru | --r)
++    ac_prev=runstatedir ;;
++  -runstatedir=* | --runstatedir=* | --runstatedi=* | --runstated=* \
++  | --runstate=* | --runstat=* | --runsta=* | --runst=* | --runs=* \
++  | --run=* | --ru=* | --r=*)
++    runstatedir=$ac_optarg ;;
++
+   -sbindir | --sbindir | --sbindi | --sbind | --sbin | --sbi | --sb)
+     ac_prev=sbindir ;;
+   -sbindir=* | --sbindir=* | --sbindi=* | --sbind=* | --sbin=* \
+@@ -1443,7 +1456,7 @@ fi
+ for ac_var in	exec_prefix prefix bindir sbindir libexecdir datarootdir \
+ 		datadir sysconfdir sharedstatedir localstatedir includedir \
+ 		oldincludedir docdir infodir htmldir dvidir pdfdir psdir \
+-		libdir localedir mandir
++		libdir localedir mandir runstatedir
+ do
+   eval ac_val=\$$ac_var
+   # Remove trailing slashes.
+@@ -1596,6 +1609,7 @@ Fine tuning of the installation directories:
+   --sysconfdir=DIR        read-only single-machine data [PREFIX/etc]
+   --sharedstatedir=DIR    modifiable architecture-independent data [PREFIX/com]
+   --localstatedir=DIR     modifiable single-machine data [PREFIX/var]
++  --runstatedir=DIR       modifiable per-process data [LOCALSTATEDIR/run]
+   --libdir=DIR            object code libraries [EPREFIX/lib]
+   --includedir=DIR        C header files [PREFIX/include]
+   --oldincludedir=DIR     C header files for non-gcc [/usr/include]
+@@ -1633,6 +1647,8 @@ Optional Features:
+   --disable-silent-rules  disable silent building
+   --enable-static_link    use this to link the tools to their libraries
+                           statically (default is dynamic linking
++  --enable-shared         use this to enable dynamic linking of tools and
++                          libraries [default=yes]
+   --disable-thin_check_needs_check
+                           required if thin_check version is < 0.3.0
+   --disable-cache_check_needs_check
+@@ -7842,6 +7858,8 @@ main ()
+     if (*(data + i) != *(data3 + i))
+       return 14;
+   close (fd);
++  free (data);
++  free (data3);
```

Comment:
Does this mean there was a memory leak upstream? Not sure I like this being silently in the middle of allow-static-build patch…

Reply:
`configure` (the output from autoconf) is commited in the repository. I ran autoconf, with a slightly more recent version than last maintainer of that file did (more than a year ago).
It looks like it's code that tests features from the environment to detect them, as autoconf usually does. This looks like fixes further upstream in autoconf.

Just like the `runstate` parameter additions, I think they can safely be ignored.